### ISSUE TITLE
Issue with current injection via ACSource in Brian

### ISSUE THREAD

**Issue by appukuttan-shailesh** [NeuralEnsemble/PyNN#442]
There is a problem with the output of current injection via ACSource in Brian. The current profile is incorrect (as can be inferred from the voltage trace in below figure), and also the current injection fails to cease at t_stop.

The figure below illustrates this (for this [source code](https://github.com/appukuttan-shailesh/PyNN_testData/blob/master/testCase_AC.py)):

Brian Output:
![ac_brian_bad](https://cloud.githubusercontent.com/assets/24866517/22545323/8e7a08c6-e938-11e6-8eac-3f894b06d783.png)

NEURON Output:
![ac_neuron_good](https://cloud.githubusercontent.com/assets/24866517/22545331/9a6132ae-e938-11e6-97dc-e578c4a87bb7.png)

**Comment by appukuttan-shailesh** [comment]
The problems here are similar to those described in issue #437. The solution for ensuring that current injection stops at t_stop is same as that discussed [there](https://github.com/NeuralEnsemble/PyNN/issues/437#issuecomment-276927756).

Brian requires time to be specified with units of seconds. Most of the required conversion is done, except for simulator.state.dt which is still specified in ms, and the frequency which is converted to 1/ms. To fix this, we need to update the following:

i) method _generate() in line https://github.com/NeuralEnsemble/PyNN/blob/master/pyNN/brian/standardmodels/electrodes.py#L114
from:
``` python
self.times = numpy.arange(self.start, self.stop, simulator.state.dt)
```
to:
``` python
self.times = numpy.arange(self.start, self.stop, simulator.state.dt * 1e-3)
```

ii) method _compute() in line https://github.com/NeuralEnsemble/PyNN/blob/master/pyNN/brian/standardmodels/electrodes.py#L117
from:
``` python
return self.amplitude * numpy.sin(time * 2 * numpy.pi * self.frequency / 1000. + 2 * numpy.pi * self.phase / 360)
```
to:
``` python
return self.amplitude * numpy.sin(time * 2 * numpy.pi * self.frequency + 2 * numpy.pi * self.phase / 360)
```

Brian output with above fixes:
---------------------------------------
![ac_brian_fixed](https://cloud.githubusercontent.com/assets/24866517/22547244/17feb634-e941-11e6-84a1-4f4168bccfe5.png)

**Comment by appukuttan-shailesh** [comment]
If the **alternative** fix for problem 1 in issue #437 is adopted, then the method '_generate' in line https://github.com/NeuralEnsemble/PyNN/blob/master/pyNN/brian/standardmodels/electrodes.py#L114 should be changed from:
``` python
def _generate(self):
        self.times = numpy.arange(self.start, self.stop, simulator.state.dt)
```
to
``` python
def _generate(self):
    self.times = numpy.arange(self.start, self.stop, simulator.state.dt * 1e-3)
    self.times = numpy.append(self.times, self.stop)
```
_Note: Issue #445 would need to be fixed for this to work properly._

**Comment by appukuttan-shailesh** [comment]
Found two more issues with current injection via AC source, as seen in the below outputs:
_Note: start = 47.5 ms and offset = 0.1_

NEURON output:
![neuron_good_stop47 5_offset0 1](https://cloud.githubusercontent.com/assets/24866517/22733865/daeb2b80-edf3-11e6-9f40-210c302a2586.png)

Brian output:
![brian_bad_stop47 5_offset0 1](https://cloud.githubusercontent.com/assets/24866517/22733868/df351886-edf3-11e6-898c-ca9f370362e7.png)

The issues are:
1) Brian implementation evaluates current amplitudes with times from t = 0 ms, instead of from t = t_start
2) Brian implementation is not taking into consideration the parameter 'offset' while evaluating the amplitudes

Combined fix for all of the above
-------------------------------------------
Methods _generate() and _compute() for the class _ACSource_  in file https://github.com/NeuralEnsemble/PyNN/blob/master/pyNN/brian/standardmodels/electrodes.py
needs to be updated to:
``` python
def _generate(self):
        self.times = numpy.arange(self.start, self.stop + simulator.state.dt * 1e-3, simulator.state.dt * 1e-3)

def _compute(self, time):
        return self.offset + self.amplitude * numpy.sin((time-self.start) * 2 * numpy.pi * self.frequency + 2 * numpy.pi * self.phase / 360)
```

Brian output with above fix:
![brian_fixed_stop47 5_offset0 1](https://cloud.githubusercontent.com/assets/24866517/22734065/aa7c8678-edf4-11e6-9648-50256de19e66.png)

**Comment by apdavison** [comment]
Fixed in #467
